core/src/main/scala/org/http4s/rho/RouteExecutable.scala

```diff
@@ -17,22 +17,18 @@ trait RouteExecutable[T <: HList] extends TypedBuilder[T] {
   /** [[Method]] of the incoming HTTP [[Request]] */
   def method: Method
 
-  /** Create a [[RhoAction]] from this [[RouteExecutable]] with the provided converters */
-  def makeAction[F](f: F, hf: HListToFunc[T, F]): RhoAction[T, F]
+//  /** Create a [[RhoAction]] from this [[RouteExecutable]] with the provided converters */
+  def makeAction(action: Action[T]): RhoRoute[T]
 
   /** Compiles a HTTP request definition into an action */
-  final def |>>[F, R](f: F)(implicit hf: HListToFunc[T, F], srvc: CompileService[F, R]): R =
-    srvc.compile(makeAction(f, hf))
+  final def |>>[R](action: Action[T])(implicit srvc: CompileService[R]): R =
```

Comment:
I'm not sure I agree that `Action` should be viewed as an implementation detail: with a concrete type like `Action` we can manipulate it: make before filters etc.